src/background.js

```diff
@@ -4,98 +4,117 @@
-      });
-    }
-  });
+/**
+ * Track the active window
+ */
+function trackActiveWindow() {
+    chrome.tabs.onActivated.addListener(storeActiveTab);
+    chrome.windows.onFocusChanged.addListener(windowId => {
+        if (windowId !== chrome.windows.WINDOW_ID_NONE) {
+            storeActiveTab({ windowId });
+        }
+    });
 }
 
-// I like to move it!
-function moveIt(tab, event) {
-  console.log(tab.windowId + ' - chrome.tabs.' + event + ' - get new tab.index: ' + tab.index);
-  console.log(tab.windowId + ' - chrome.tabs.' + event + ' - get currentIndex: ' + currentIndex[tab.windowId]);
-  if (Number.isInteger(currentIndex[tab.windowId])) {
-    const moveToIndex = currentIndex[tab.windowId] + 1;
-
-    // Current is where it will move :-)
-    currentIndex[tab.windowId] = moveToIndex;
-
-    // Nothing to do
-    if (tab.index === moveToIndex) {
-      console.log(tab.windowId + ' - chrome.tabs.' + event + ' - tab.index: ' + tab.index + ' == moveToIndex: ' + moveToIndex + ' (nothing to do)');
-      return;
+/**
+ * Query and store the active tab
+ *
+ * @param {*} {windowId} Optional object containing windowId
+ */
+function storeActiveTab({ windowId } = {}) {
+    if (logErrorExisting("storeActiveTab")) {
+        return;
     }
+    // get the active tab
+    chrome.tabs.query(
+        {
+            active: true,
+            lastFocusedWindow: true,
+            windowType: "normal",
+            windowId
+        },
+        ({ 0: { id, windowId: activeWindowId, index, openerTabId } }) => {
+            console.log(`Active tab: ${id}, opened by tab: ${openerTabId}`);
+            // and store it's information
+            chrome.storage.local.set({
+                [ACTIVE_TAB_KEY]: { tabId: id, windowId: activeWindowId, tabIndex: index, parentTabId: openerTabId }
+            });
+        }
+    );
+}
 
-    // Move!
-    chrome.tabs.move(tab.id, {
-      index: moveToIndex
+/**
+ * When a new tab is created, track its position as well
+ */
+function startUpdateTabPositions() {
+    chrome.tabs.onCreated.addListener(newTab => {
+        chrome.storage.local.get(ACTIVE_TAB_KEY, ({ [ACTIVE_TAB_KEY]: { tabId, windowId, tabIndex } }) => {
+            if (logErrorExisting("startUpdateTabPosition")) {
+                return;
+            }
+            if (newTab.windowId === windowId) {
+                chrome.tabs.move(newTab.id, { index: tabIndex + 1 });
+                console.log(`Moved new tab: ${newTab.id} next to ${tabId}`);
+            }
+        });
     });
-    console.log(tab.windowId + ' - chrome.tabs.' + event + ' - move to: ' + moveToIndex);
-  }
 }
 
 /**
- * Move the new tab after the current one.
+ * When a tab is closed, attempt to go back to its parent
+ * if possible
  */
-// On created
-chrome.tabs.onCreated.addListener(tab => {
-  chrome.tabs.onMoved.removeListener(eventOnMoved);
-  moveIt(tab, 'onCreated');
-  chrome.tabs.onMoved.addListener(eventOnMoved);
-});
+function startUpdateParentPositions() {
```

Comment:
This is where we added support for returning to parent after tab is closed.

Doesn't fully work because chrome does not retain that information after the tab has been navigated away from